ESP-Ruuvi-Collector.ino

```diff
@@ -1,97 +1,142 @@
+const char INFLUX_IP[] = "192.168.x.y";
+const char INFLUX_USER[] = "some";
+const char INFLUX_PASS[] = "user";
+const char INFLUX_MEASUREMENT[] = "ruuviTag";
+
+char formatStringFive[] = "temperature=%0.3f,pressure=%i,humidity=%0.3f,accel_x=%i,accel_y=%i,accel_z=%i,milivolts=%i";
+char formatStringThree[] = "temperature=%0.3f,pressure=%i,humidity=%i,milivolts=%i";
+
+// 16 bit
+short getShort(byte* data, int index)
+{
+  return (short)((data[index] << 8) + (data[index + 1]));
+}
+
+short getShortone(byte* data, int index)
+{
+  return (short)((data[index]));
+}
+
+// 16 bit
+unsigned short getUShort(byte* data, int index)
+{
+  return (unsigned short)((data[index] << 8) + (data[index + 1]));
+}
+
+unsigned short getUShortone(byte* data, int index)
+{
+  return (unsigned short)((data[index]));
+}
+
+void DecodeV3(byte* data)
+{
+  digitalWrite(22, HIGH);
+  short tempRaw = getShortone(data, 4) - 2;
+  short tempRawdec = getUShortone(data, 5);
+  double temperature = (double)tempRaw + (double)tempRawdec / 100;
+
+  byte humRaw = getUShortone(data, 3);
+  short humidity = humRaw / 2;
+  unsigned int pressure = (getUShort(data, 6) + 50000);
+  unsigned int voltageraw = getUShort(data, 14);
+  short voltage = (short)voltageraw;
+
+  char fields[256];
+  sprintf(fields, formatStringThree, temperature, pressure, humidity, voltage);
+  bool writeSuccessful = influx.write(INFLUX_MEASUREMENT, "device=esp", fields);
+  if (!writeSuccessful)
+  {
+    Serial.print("error: ");
+    Serial.println(influx.getResponse());
+  }
+ 
+}
+
+  void DecodeV5(byte* data)
+  {
+  digitalWrite(22, LOW);
+  short tempRaw = getShort(data, 3);
+  double temperature = (double)tempRaw * 0.005;
+  unsigned short humRaw = getUShort(data, 5);
+  double humidity = (double)humRaw * 0.0025;
+  unsigned int pressure = (getUShort(data, 7) + 50000);
+  short accelX = getShort(data, 9);
+  short accelY = getShort(data, 11);
+  short accelZ = getShort(data, 13);
+
+  unsigned short voltRaw = data[15] << 3 | data[16] >> 5;
+  unsigned char tPowRaw = data[16] && 0x1F;
+  unsigned short voltage = voltRaw + 1600;
+  char power = tPowRaw* 2 - 40;
+
+  char fields[256];
+  sprintf(fields, formatStringFive, temperature, pressure, humidity, accelX, accelY, accelZ, voltage);
+  bool writeSuccessful = influx.write(INFLUX_MEASUREMENT, "device=esp", fields);
+  if (!writeSuccessful)
+  {
+    Serial.print("error: ");
+    Serial.println(influx.getResponse());
+  }
+}
+
+class AdvertisedDeviceCallbacks: public BLEAdvertisedDeviceCallbacks {
+    void onResult(BLEAdvertisedDevice advertisedDevice) {
+      byte* mData = (byte*)advertisedDevice.getManufacturerData().data();
+
+      if (mData[0] == 0x99 && mData[1] == 0x04 && mData[2] == 0x03)
+      {
+       /* Serial.print (mData[0]);
+        Serial.print (" ; ");
+        Serial.print (mData[1]);
+        Serial.print (" ; ");
+        Serial.print (mData[2]);
+        Serial.println (" ; ");
+        */
+        DecodeV3(mData);

```

Comment:
I think you just removed support for v5 protocol here. You could leave if clause like it was (because for both first two bytes are 0x99 0x04) and inside check for third byte and run DecodeV3 for 0x03 or DecodeV5 for 0x05.

Reply:
Not sure, why the pull did not include the second if clause on 0x05. Have to check, how to update a pullrequest. ;)

Reply:
> Just add another commit to your branch, it will update itself 😉 W dniu sob., 9.02.2019 o 01:13 Till <notifications@github.com> napisał(a):
> […](#)
> *@Quas7* commented on this pull request. ------------------------------ In ESP-Ruuvi-Collector.ino <[#2 (comment)](https://github.com/domints/ESP-Ruuvi-Collector/pull/2#discussion_r255273154)> : > +} + +class AdvertisedDeviceCallbacks: public BLEAdvertisedDeviceCallbacks { + void onResult(BLEAdvertisedDevice advertisedDevice) { + byte* mData = (byte*)advertisedDevice.getManufacturerData().data(); + + if (mData[0] == 0x99 && mData[1] == 0x04 && mData[2] == 0x03) + { + /* Serial.print (mData[0]); + Serial.print (" ; "); + Serial.print (mData[1]); + Serial.print (" ; "); + Serial.print (mData[2]); + Serial.println (" ; "); + */ + DecodeV3(mData); Not sure, why the pull did not include the second if clause on 0x05. Have to check, how to update a pullrequest. ;) — You are receiving this because you commented. Reply to this email directly, view it on GitHub <[#2 (comment)](https://github.com/domints/ESP-Ruuvi-Collector/pull/2#discussion_r255273154)>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABC54SdCWATiOouYa43quZARb_rjqVZCks5vLhLEgaJpZM4aqYM1> .

I already committed directly to my master branch - should this not do the trick?